### DISCUSSION TITLE
Next 15 unclear build error on Vercel

### DISCUSSION THREAD

**Discussion by elie222** [vercel/next.js#72107] [App Router]
I'm getting a build error although it's unclear why.
Everything worked fine on Next 14.
The build also works locally but fails on Vercel.

This is the repo:
https://github.com/elie222/inbox-zero

This is the error:

```
inbox-zero-ai:build:    Creating an optimized production build ...
inbox-zero-ai:build:  ✓ (serwist) Bundling the service worker script with the URL '/sw.js' and the scope '/'...
inbox-zero-ai:build:  ✓ Compiled successfully
inbox-zero-ai:build:    Linting and checking validity of types ...
inbox-zero-ai:build: It looks like you're trying to use TypeScript but do not have the required package(s) installed.
inbox-zero-ai:build: 
inbox-zero-ai:build: Please install @types/react by running:
inbox-zero-ai:build: 
inbox-zero-ai:build: 	pnpm install --save-dev @types/react
inbox-zero-ai:build: 
inbox-zero-ai:build: If you are not trying to use TypeScript, please remove the tsconfig.json file from your package root (and any TypeScript files in your app and pages directories).
inbox-zero-ai:build: 
inbox-zero-ai:build:  ELIFECYCLE  Command failed with exit code 1.
```

I do have `@types/react` and `typescript` installed as can be seen here: https://github.com/elie222/inbox-zero/blob/next15/apps/web/package.json

**Comment by In-S4n3** [comment]
I have exactly the same error... it builds locally without any problem but it does't in vercel...

**Reply by elie222** [reply]
Any luck on getting this fixed?

**Comment by damien-schneider** [comment]
If you’ve encountered the same issue I did, it was due to having a monorepo with multiple React dependencies. Specifically, one of my packages was still using React 18, which caused a type mismatch.
